### PR TITLE
e2e: test migrate command

### PR DESCRIPTION
**Pull request by gyuho**
Fix https://github.com/coreos/etcd/issues/5407.

### CODE REVIEW

**Comment by gyuho** [comment]
/cc @xiang90 @heyitsanthony 

Thanks!

**Comment by gyuho** [comment]
Need to add more checks (esp. if v3 writes after migrations are continuous from v2 revisions)/

**Comment by gyuho** [comment]
@xiang90 @heyitsanthony PTAL.

File: e2e/ctl_v3_migrate_test.go [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+	keys := make([]string, 3)
+	vals := make([]string, 3)
+	for i := range keys {
+		keys[i] = fmt.Sprintf("foo_%d", i)
+		vals[i] = fmt.Sprintf("bar_%d", i)
+	}
+
+	// 2. write keys to v2 store
+	var v2Resps v2ResponseByCreation
```
Comment (xiang90) [inline e2e/ctl_v3_migrate_test.go]: i think the get-json and sort-response thing are over kill. We know what we put into v2. We just need to compare what we get from v3 after migration against what we put into v2. We do not need to compare it with the responses we get out of v2.

Comment (gyuho) [inline e2e/ctl_v3_migrate_test.go]: ok will fix.

**Comment by gyuho** [comment]
@xiang90 PTAL.

Thanks.

File: e2e/etcd_test.go [hunk 2]
```diff
@@ -344,6 +344,56 @@ func (cfg *etcdProcessClusterConfig) tlsArgs() (args []string) {
...
+		if err := <-readyC; err != nil {
+			epc.Close()
+			return err
+		}
+	}
+	return nil
+}
+
+func (epc *etcdProcessCluster) Stop() (err error) {
+	for _, p := range epc.procs {
```
Comment (xiang90) [inline e2e/etcd_test.go]:
```go
errors := []string
... append error if any
return error.New(strings.Join(errors,","))
```

**Comment by gyuho** [comment]
Need to make new functions less redundant (WIP)

**Comment by gyuho** [comment]
@xiang90 PTAL. Made it reuse as much code as possible.
And simplified migrate test.

**Comment by gyuho** [comment]
@xiang90 Any objection on this?

**Comment by gyuho** [comment]
@xiang90 All addressed. PTAL.

File: e2e/etcd_test.go [hunk 3]
```diff
@@ -344,12 +329,54 @@ func (cfg *etcdProcessClusterConfig) tlsArgs() (args []string) {
 	return args
 }
 
-func (epc *etcdProcessCluster) Close() (err error) {
+func (epc *etcdProcessCluster) launch() (err error) {
```
Comment (gyuho) [inline e2e/etcd_test.go]: will change this to `Start`

**Comment by xiang90** [comment]
lgtm

**Comment by gyuho** [comment]
thanks. merging in greens.

**Pull request merged**